### DISCUSSION TITLE
unable to obtain certificates

### DISCUSSION THREAD

**Discussion by coffseducation** [drakkan/sftpgo#1519] [General]
Hello!
I have setup SFTPgo on a home server for backups. I have it backing up my VPS running Caprover using Dupliciti.
It is working well!

I am using aapanel on the home backup server and finally got ssl certs to work.
I tried adding sftp/port 8080 to the domainto utulise the ssl but aapanel complained it was outside of the port range.
A dev said it could not be added.

I found ACME in the SFTPgo settings but get:

unable to obtain certificates: error: one or more domains had a problem: [backup.mydomain.academy] [backup.mydomaint.academy] acme: error presenting token: could not start HTTP server for challenge: listen tcp :80: bind: address already in use 

I installed sftpgo as per defaults using port 8080 (Ubuntu ppa) - but it does not look possible to add an ssl!

I do not want to have an insecure connection. I am backing up some sensitive data. How can I create an ssl?
Is there another (easy) way? I am a server noob.
Can I change to port 80?

I was hoping sftpgo would make things easier for me as a server noob, but perhaps I should just use regular sftp to avoid this complication?

Thanks.

**Comment by drakkan** [comment]
> unable to obtain certificates: error: one or more domains had a problem: [backup.mydomain.academy] [backup.mydomaint.academy] acme: error presenting token: could not start HTTP server for challenge: listen tcp :80: bind: address already in use

to complete tha ACME challenge SFTPGo needs to listen on port 80 and it seems already in use. 
If you already have an HTTP server you can take a look at the docs [here](https://github.com/drakkan/sftpgo/blob/main/docs/howto/lets-encrypt-certificate.md)

**Reply by coffseducation** [reply]
Ok, thanks for the reply. I have looked at that link but cannot comprehend it much, I'm sorry.

re: "80 and it seems already in use"
- I set up a web page trying to get ssl working. If I delete this, will that mean it is no longer in use and sftpgo can use it? Are other changes needed?
Thanks
![image](https://github.com/drakkan/sftpgo/assets/99161534/c9dcc1fb-728b-4b48-91f7-131c3ceb676b)

**Reply by drakkan** [reply]
Yes, if no other service listen on port 80 SFTPGo will complete the ACME challenge and obtain a certificate.
The domain you request the certificate for must resolve back to the IP address where SFTPGo listens.
